### PR TITLE
Add type hints to `management`

### LINKED ISSUE #470: Add type hints
As the project is now supporting Python3.7 and greater, we could add type hints. I'd be happy to work on this if this is an option.

For reference: https://typing.readthedocs.io/en/latest/source/libraries.html

Tracked PRs:

- https://github.com/auth0/auth0-python/pull/472
- https://github.com/auth0/auth0-python/pull/497

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #470.

This is ready for review, but I might take a second look, probably forgot a few things.

### CODE REVIEW

**Comment by adamjmcgrath** [comment]
lgtm - thanks @Viicos

Just need to fix the build

**Comment by Viicos** [comment]
@adamjmcgrath done, sorry for the delay

**Review by adamjmcgrath — APPROVED** [review]

File: auth0/management/users.py [hunk 1]
```diff
@@ -27,37 +33,37 @@ class Users:
...
+        page: int = 0,
+        per_page: int = 25,
+        sort: str | None = None,
+        connection: str | None = None,
+        q: str | None = None,
+        search_engine: str | None = None,
+        include_totals: bool = True,
+        fields: List[str] | None = None,
+        include_fields: bool = True,
+    ) -> List[dict[str, Any]]:
```
Comment (lilidworkin) [inline auth0/management/users.py:66]: I believe this is the wrong type? I think it should be `Dict[str, Any]`

File: auth0/management/organizations.py [hunk 2]
```diff
@@ -219,13 +240,13 @@ def delete_organization_connection(self, id, connection_id):
...
-        from_param=None,
-        take=None,
-    ):
+        id: str,
+        page: int | None = None,
+        per_page: int | None = None,
+        include_totals: bool = True,
+        from_param: str | None = None,
+        take: int | None = None,
+    ) -> list[dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (adamjmcgrath) [reply auth0/management/organizations.py:249]: @shchotse This PR is merged, would you mind creating a separate issue? Thanks

File: auth0/management/users.py (hunk 1 above)
Comment (Viicos) [reply auth0/management/users.py:66]: @lilidworkin thanks. See https://github.com/auth0/auth0-python/pull/497#discussion_r1282096676 for more details

**Comment by michaeloliverx** [comment]
Are there any plans to add some more concrete return types? E.g. using `TypedDict`


It would be a huge win for DX, currently I am maintaining local definitions and doing this:
```python
from typing import NotRequired, TypedDict


class Role(TypedDict):
    id: str
    name: str
    description: str


class RolesResponse(TypedDict):
    """https://auth0.com/docs/api/management/v2/roles/get-roles"""
    start: int
    limit: int
    total: int
    roles: list[Role]

...

user_roles: RolesResponse = auth0_client.users.list_roles(user_id)
```

**Pull request merged**